[reland2][inductor] Add an AOT compilation mode for Inductor CPP backend

desertfire opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #96654
* __->__ #96520

Summary: This is a reland of https://github.com/pytorch/pytorch/pull/94822.
Solved the long compilation issue for inductor cpp tests.

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10

desertfire committed:
[reland2][inductor] Add an AOT compilation mode for Inductor CPP backend

Summary: This is a reland of https://github.com/pytorch/pytorch/pull/94822.
Solved the long compilation issue for inductor cpp tests.

[ghstack-poisoned]

huydhn reviewed (approved):
LGTM! I double check the TTS in the PR, and it looks normal inclusing ASAN and macos TTS

desertfire committed:
Update on "[reland2][inductor] Add an AOT compilation mode for Inductor CPP backend"


Summary: This is a reland of https://github.com/pytorch/pytorch/pull/94822.
Solved the long compilation issue for inductor cpp tests.

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10

[ghstack-poisoned]

desertfire commented:
@pytorchbot merge
